File: 3sem/4lab/Banks/models/BankConfigurations/BankConfiguration.cs
namespace Banks.Models.BankConfigurations;

public record BankConfiguration
{
    public BankConfiguration(
        CreditBankAccountConfiguration creditConfiguration,
        DebitBankAccountConfiguration debitConfiguration,
        DepositBankAccountConfiguration depositConfiguration)
    {
        ArgumentNullException.ThrowIfNull(creditConfiguration);
        ArgumentNullException.ThrowIfNull(debitConfiguration);
        ArgumentNullException.ThrowIfNull(depositConfiguration);

        CreditConfiguration = creditConfiguration;
        DebitConfiguration = debitConfiguration;
        DepositConfiguration = depositConfiguration;
    }

    public CreditBankAccountConfiguration CreditConfiguration { get; }
    public DebitBankAccountConfiguration DebitConfiguration { get; }
    public DepositBankAccountConfiguration DepositConfiguration { get; }
}

File: 4sem/1lab/src/main/java/BanksRemake/Domain/Models/BankConfigurations/BankConfiguration.java
package banksRemake.domain.models.bankConfigurations;

/**
 * A set of parameters characterizing the bank products.
 * @param creditBankAccountConfiguration a set of parameters characterizing the credit account.
 * @param debitBankAccountConfiguration a set of parameters characterizing the debit account.
 * @param depositBankAccountConfiguration a set of parameters characterizing the deposit account.
 */
public record BankConfiguration(CreditBankAccountConfiguration creditBankAccountConfiguration,
                                DebitBankAccountConfiguration debitBankAccountConfiguration,
                                DepositBankAccountConfiguration depositBankAccountConfiguration) {
}